app/assets/javascripts/components/option-select.js

```diff
@@ -123,38 +135,42 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
      * We do this in the JavaScript because if the JavaScript is not active then the button shouldn't
      * be there as there is no JS to handle the click event.
     */
-    var $containerHead = this.$optionSelect.find('.js-container-button')
-    var jsContainerHeadHTML = $containerHead.html()
+    var containerHead = this.$optionSelect.querySelector('.js-container-button')
+    var jsContainerHeadHTML = containerHead.innerHTML
 
     // Create button and replace the preexisting html with the button.
-    var $button = $('<button>')
-    $button.addClass('js-container-button app-c-option-select__title app-c-option-select__button')
+    var button = document.createElement('button')
+    button.setAttribute('class', 'js-container-button app-c-option-select__title app-c-option-select__button')
     // Add type button to override default type submit when this component is used within a form
-    $button.attr('type', 'button')
-    $button.attr('aria-expanded', true)
-    $button.attr('id', $containerHead.attr('id'))
-    $button.attr('aria-controls', this.$optionsContainer.attr('id'))
-    $button.html(jsContainerHeadHTML)
-    $containerHead.replaceWith($button)
+    button.setAttribute('type', 'button')
+    button.setAttribute('aria-expanded', true)
+    button.setAttribute('id', containerHead.getAttribute('id'))
+    button.setAttribute('aria-controls', this.$optionsContainer.getAttribute('id'))
+    button.innerHTML = jsContainerHeadHTML
+    containerHead.parentNode.replaceChild(button, containerHead)
   }
 
   OptionSelect.prototype.attachCheckedCounter = function attachCheckedCounter (checkedString) {
-    this.$optionSelect.find('.js-container-button')
-      .after('<div class="app-c-option-select__selected-counter js-selected-counter">' + checkedString + '</div>')
+    var element = document.createElement('div')
+    element.setAttribute('class', 'app-c-option-select__selected-counter js-selected-counter')
+    element.innerHTML = checkedString
+    this.$optionSelect.querySelector('.js-container-button').insertAdjacentElement('afterend', element)
   }
 
   OptionSelect.prototype.updateCheckedCount = function updateCheckedCount () {
     var checkedString = this.checkedString()
-    var checkedStringElement = this.$optionSelect.find('.js-selected-counter')
+    var checkedStringElement = this.$optionSelect.querySelector('.js-selected-counter')
 
     if (checkedString) {
-      if (checkedStringElement.length) {
-        checkedStringElement.text(checkedString)
-      } else {
+      if (checkedStringElement == null) {
         this.attachCheckedCounter(checkedString)
+      } else {
+        checkedStringElement.textContent = checkedString
       }
     } else {
-      checkedStringElement.remove()
+      if (checkedStringElement) {
```

Comment:
you can combine this in `} else if (checkedStringElement) {`

Reply:
Fixed - thanks